src/services/examplesService.ts

```diff
@@ -0,0 +1,116 @@
+import axios from 'axios';
+
+import { SERVER_URL } from '@constants/auth';
+
+import authService from './authService';
+
+interface GistFile {
+  content: string;
+}
+
+interface UploadCustomExampleData {
+  code: string;
+  description: string;
+  exampleName: string;
+}
+
+const uploadExample = async (data: UploadCustomExampleData) => {
+  const jwtToken = await authService.getJwtToken();
+
+  const files: { [key: string]: GistFile } = {
+    'snippet.tsx': {
+      content: data.code,
+    },
+    'name.txt': {
+      content: data.exampleName,
+    },
+  };
+
+  const body = {
+    name: data.exampleName,
+    description: data.description,
+    files,
+    publicGist: true,
+  };
+
+  const response = await axios.post(`${SERVER_URL}/gists`, body, {
```

Comment:
should we add authorization token to each request in case we have something similar in the axios interceptors

Reply:
Done